sdk/core/azure-core/azure/core/rest/_requests_basic.py

```diff
@@ -85,10 +85,11 @@ def encoding(self, value):
         self._encoding = value
         self._internal_response.encoding = value
 
-    @property
-    def text(self):
+    def text(self, encoding=None):
         # this will trigger errors if response is not read in
         self.content  # pylint: disable=pointless-statement
+        if encoding:
+            self._internal_response.encoding = encoding
```

Comment:
Should we update _internal_response.encoding in text method?

This will cause some side effect?

Reply:
@xiangyan99 talked to @lmazuel offline, and we agreed to not handle requests text and encoding as a special case, so all of this code is gone now!